### PR TITLE
feat: Adds stepper (progress) component

### PR DESCRIPTION
**Pull request by schne324**
This patch implements the "light theme" stepper. Support for the dark theme will be added in a separate PR (which we can then use in the extensions IGT progress thing)

### CODE REVIEW

File: packages/react/src/components/Stepper/index.tsx [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+import TooltipTabstop from '../TooltipTabstop';
+
+interface BaseStepProps {
+  status: 'current' | 'complete' | 'future';
+  className?: string;
+}
+
+interface StepWithChildren extends BaseStepProps {
+  children: React.ReactNode;
+  tooltip?: never;
```
Comment (schne324) [inline packages/react/src/components/Stepper/index.tsx:13]: @scurker I forgot to circle back and try out your suggestion here for replacing `never` here with something like: 

```
function propsWithTooltip(props: StepWithChildren | StepWithTooltip): props is StepWithTooltip {
    return !!(props as StepWithTooltip).tooltip
}
```

File: packages/styles/stepper.css [hunk 2]
```diff
@@ -0,0 +1,87 @@
+.Stepper {
+  --step-indicator-size: 25px;
+  --step-line-height: 8px;
+  --step-line-offset: 18px;
```
Comment (scurker) [inline packages/styles/stepper.css:4]: No vars? Wouldn't we at least want to make these relative to the font size?

**Review by scurker — COMMENTED** [review]
Just one question, code LGTM otherwise

File: packages/styles/stepper.css (hunk 2 above)
Comment (schne324) [reply packages/styles/stepper.css]: Made the change to `em` -- good call 👍

**Review by scurker — APPROVED** [review]

**Pull request merged**